lib/storage/bucket.js

```diff
@@ -252,16 +274,18 @@ Bucket.prototype.getFiles = function(query, callback) {
       callback(err);
       return;
     }
+
     var files = (resp.items || []).map(function(item) {
       var file = that.file(item.name);
       file.metadata = item;
       return file;
     });
+    var prefixes = resp.prefixes || null;
     var nextQuery = null;
     if (resp.nextPageToken) {
       nextQuery = extend({}, query, { pageToken: resp.nextPageToken });
     }
-    callback(null, files, nextQuery);
+    callback(null, files, nextQuery, prefixes);
```

Comment:
For me the best solution is to return object with files, prefixes instead files but I'm afraid that it's too big change, so I proposed it in this way.